### PR TITLE
Initial API Tests

### PR DESCRIPTION
**Pull request by joshgarde**
Implemented API testing from the transport layer (HTTP) down to an in-memory database (H2), covering all of the API's current data queries and mutations. Also contained in this PR is a standardization of the database's SQL schema to support both PostgreSQL and H2 bootstrapping. Several hotfixes to API bugs were also implemented as they were discovered in the testing process. The test and coverage reports are available below.

[test-report.zip](https://github.com/podaac/swodlr/files/9590529/test-report.zip)
[coverage-report.zip](https://github.com/podaac/swodlr/files/9590531/coverage-report.zip)

### CODE REVIEW

File: src/test/java/gov/nasa/podaac/swodlr/L2RasterProductTests.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                var timestamp = timestamps.get(0);
+                assertTrue(timestamp.compareTo(start) >= 0, "timestamp: %s, start: %s".formatted(timestamp, start));
+            });
+
+        // State
+        response
+            .path("createL2RasterProduct.status[*].state")
+            .entityList(String.class)
+            .hasSize(1)
+            .containsExactly("NEW");
```
Comment (frankinspace) [inline src/test/java/gov/nasa/podaac/swodlr/L2RasterProductTests.java:95]:
```suggestion
            .containsExactly(State.NEW.toString());
```

**Comment by frankinspace** [comment]
This is awesome, nice work!

**Comment by joshgarde** [comment]
Updated test results with changes

[test-results.zip](https://github.com/podaac/swodlr/files/9649842/test-results.zip)

**Comment by joshgarde** [comment]
Latest test results with updates
[test-results.zip](https://github.com/podaac/swodlr/files/9649866/test-results.zip)

**Review by frankinspace — APPROVED** [review]

**Pull request merged**